### ISSUE TITLE
Clicking person's name in forwarding dialog loses my message

### ISSUE THREAD

**Issue by andybalaam** [vector-im/element-web#20636]
### Steps to reproduce

1. In a room
2. Hover over a message and choose Forward in the context menu
3. Click the name of a person in the Forward Message dialog

![rec-2022-01-19_11 35 38](https://user-images.githubusercontent.com/76812/150122465-8045d7c8-c3b5-40fd-8846-8209808f8a03.gif)

#### What did you expect?
I expected the message to be forwarded to the person I clicked, or nothing to happen

#### What happened instead?
I lost the message I was trying to forward, and jumped to looking at the person I clicked on

### URL for webapp

https://develop.element.io

### Application version

Element version: 840867bc0212-react-8427bf46efbd-js-a50a627300df Olm version: 3.2.8

### Will you send logs?

No

**Comment by HarHarLinks** [comment]
while I understand that frustration, I think having the feature of navigating to that room from the forwarding UI at all is great. I use it frequently after forwarding to open said room.

Maybe it just needs to communicate better what is going to happen when you click certain things?

The demonstrated case could be bandaged by placing the message you are trying to forward in your composer upon opening the room (as long as you have not clicked "Send" yet). However that would need some variations for non-text, as e.g. file sending UI works quite differently. It is probably better to move the "switch to room" action to a more clearly labeled button instead.

**Comment by andybalaam** [comment]
Yes, I agree a clearly-labelled "Go to room" button would be useful.  The current behaviour was very confusing to me.
